ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Snackbar/SnackState.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  SnackState.swift
+//
+//
+//  Created by Filip Wiesner on 15.01.2023.
+//
+
+import Combine
+import Foundation
+import Semaphore
+import SwiftUI
+
+/**
+ Visual information for a snackbar.
+ */
+public protocol SnackVisuals: Equatable {
+    var message: String { get }
+    var actionLabel: String? { get }
+    
+    /**
+     Duration in **seconds**
+     */
+    var duration: Int { get }
+}
+
+/**
+ Cause for a snackbar disappearance.
+ */
+public enum SnackResult {
+    case actionPerformed
+    case dismissed
+}
+
+/**
+ State holder that coordinates appearence of the snacbars. Contains currentData which is used for hooking up ``SnackHost``.
+ ``SnackVisuals`` protocol defines the visual aspect of the Snackbar. ``InfoErrorSnackVisuals`` are provided by default.
+ */
+@MainActor
+public class SnackState<Visuals: SnackVisuals>: ObservableObject {
+    private let semaphore = AsyncSemaphore(value: 1)
+    @Published public private(set) var currentData: SnackData<Visuals>?
+    
+    public init() {}
+    
+    /**
+     Asynchronous function for displaying the snackbar. Suspends until the snackbar is either dismissed or action is selected.
+     Only one snackbar can be visible at one time. If called asynchronously from different places, one caller will be suspended until the firs snack is dismissed.
+     
+     - Parameters:
+        - visuals: ``SnackVisuals`` the visual information of the snackbar shown
+     
+     - Returns: The ``SnackResult`` result of the asction - either ``.dismissed`` or ``.actionPerformed`` if action was clicked.
+     */
+    public func showSnack(_ visuals: Visuals) async -> SnackResult {
```

Comment:
You can use `@discardableResult` to prevent the need of using `_ = await snackState.showSnack...`

Reply:
Thanks, changed